### PR TITLE
MSFT.CSharp: Replace empty ; with more blatant {}

### PR DESCRIPTION
**Pull request by JonHanna**
Also Remove excessive semicolons

### CODE REVIEW

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/GroupToArgsBinder.cs [hunk 1]
```diff
@@ -944,7 +944,10 @@ private bool GetNextSym(CMemberLookupResults.CMethodIterator iterator)
                 while (_HiddenTypes.Contains(_pCurrentType))
                 {
                     // Move through this type and get the next one.
-                    for (; iterator.GetCurrentType() == _pCurrentType; iterator.MoveNext(_methList.IsEmpty(), _bIterateToEndOfNsList)) ;
+                    for (; iterator.GetCurrentType() == _pCurrentType; iterator.MoveNext(_methList.IsEmpty(), _bIterateToEndOfNsList))
+                    {
+                    }
```
Comment (stephentoub) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/GroupToArgsBinder.cs]: What's wrong with the previous version?

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/ArrayType.cs [hunk 2]
```diff
@@ -20,7 +20,10 @@ internal class ArrayType : CType
         public CType GetBaseElementType()
         {
             CType type;
-            for (type = GetElementType(); type.IsArrayType(); type = type.AsArrayType().GetElementType()) ;
+            for (type = GetElementType(); type.IsArrayType(); type = type.AsArrayType().GetElementType())
+            {
+            }
```
Comment (stephentoub) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/ArrayType.cs]: Same question for all of these with ; ending a loop.  Seems fine to me.

Comment (JonHanna) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/ArrayType.cs]: Ah. I'm just getting my style-guides mixed up. Sorry.

Comment (JonHanna) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/ArrayType.cs]: The double `; ;` are still worth tidying I suppose, if very minor. I'll bring this back to just removing those.

**Pull request merged**